File: Section Reports/Preview/CustomPreview/VB.NET/CustomPreview/Reports/Catalog.vb
Public Class Catalog
	Inherits GrapeCity.ActiveReports.SectionReport
	Public Sub New()
		MyBase.New()
		InitializeComponent()
		
		Document.Name = "Catalog Report"
		
		'
	End Sub
	Private Sub PageFooter_BeforePrint(ByVal sender As Object, ByVal e As EventArgs) Handles PageFooter.BeforePrint
		Dim pg As Integer = CType(PageNumber, Integer)
		
		lblPage.Text = "Page " + pg.ToString(Globalization.CultureInfo.CurrentCulture)
		
		'
	End Sub
	Public ds As Data.OleDBDataSource
End Class

File: Section Reports/Preview/CustomPreview/C#/CustomPreview/Reports/Catalog.cs
namespace GrapeCity.ActiveReports.Samples.CustomPreview
{
	partial class Catalog : GrapeCity.ActiveReports.SectionReport
	{
		public Catalog()
		{
			InitializeComponent();
			
			Document.Name = "Catalog Report";
			
			
		}
		/// <summary>
		/// BeforePrint Event
		/// ActiveReports raises this event before the section is rendered to a page.
		/// Use this event to modify the control's values before printing.
		/// The changes made here do not effect the height of the section.
		/// </summary>
		private void PageFooter_BeforePrint(object sender, System.EventArgs eArgs)
		{
			int pg = PageNumber;
			
			lblPage.Text = "Page " + pg.ToString(System.Globalization.CultureInfo.CurrentCulture);
			
			
		}
	}
}
